pkg/volume/util/volumehelper/volumehelper.go

```diff
@@ -52,6 +52,12 @@ func GetUniqueVolumeName(pluginName, volumeName string) api.UniqueVolumeName {
 	return api.UniqueVolumeName(fmt.Sprintf("%s/%s", pluginName, volumeName))
 }
 
+// GetUniqueVolumeNameForNonAttachableVolume returns the unique volume name
+// for a non-attachable volume.
+func GetUniqueVolumeNameForNonAttachableVolume(volumePlugin volume.VolumePlugin, podNamespace, podName, podSpecName string) api.UniqueVolumeName {
+	return api.UniqueVolumeName(fmt.Sprintf("%s/%s-%s-%s", volumePlugin.GetPluginName(), podNamespace, podName, podSpecName))
```

Comment:
@pmorie, thanks for addressing this. Now I am still thinking whether this
change may affect our initial intention of avoiding operating mount/mount
for the same volume/pod. If for the same pod/volume, it is ok mount
parallel if the inner/outterspec name is different, then it should be fine.

On Wed, Jul 27, 2016 at 10:10 AM, Paul Morie <notifications@github.com>
wrote:

> In pkg/volume/util/volumehelper/volumehelper.go
> <https://github.com/kubernetes/kubernetes/pull/29673#discussion_r72480743>
> :
>
> > @@ -52,6 +52,12 @@ func GetUniqueVolumeName(pluginName, volumeName string) api.UniqueVolumeName {
> >  	return api.UniqueVolumeName(fmt.Sprintf("%s/%s", pluginName, volumeName))
> >  }
> >
> > +// GetUniqueVolumeNameForNonAttachableVolume returns the unique volume name
> > +// for a non-attachable volume.
> > +func GetUniqueVolumeNameForNonAttachableVolume(volumePlugin volume.VolumePlugin, podNamespace, podName, podSpecName string) api.UniqueVolumeName {
> > +	return api.UniqueVolumeName(fmt.Sprintf("%s/%s-%s-%s", volumePlugin.GetPluginName(), podNamespace, podName, podSpecName))
>
> @jingxu97 <https://github.com/jingxu97> I'm changing this to use the pod
> UID instead of the namespace and name to aid you in #27970
> <https://github.com/kubernetes/kubernetes/pull/27970>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29673/files/e967f0d59934d271a8eca16f6582865614008d8f#r72480743>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASSNxfEUFfXSNMgrEIUUJwFFRtyLn6B0ks5qZ5EagaJpZM4JWQGg>
> .
>



-- 
- Jing